### ISSUE TITLE
When channel_priority is strict, having an uninstalled package in pinned file will fail conda install another package

### ISSUE THREAD

**Issue by yhuai** [conda/conda#8534]
## Current Behavior

tl;dr When channel_priority is strict, if you pin an uninstalled package in pinned file, conda install another package that has not been installed will fail with an error.

### Steps to Reproduce

1. Set channel_priority to strict
```
conda config --set channel_priority strict
```
2. Pin a package that has not been installed (I am using ipython as an example)
```
echo "ipython ==7.2.0" >> /anaconda3/pkgs/channel-priority-strict-and-pinned-file/conda-meta/pinned
```

3. Install another package, e.g. requests
```
conda install --name channel-priority-strict-and-pinned-file requests

WARNING: The conda.compat module is deprecated and will be removed in a future release.
Collecting package metadata: done
Solving environment: failed

# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/anaconda3/lib/python3.7/site-packages/conda/resolve.py", line 320, in _get_strict_channel
        channel_name = self._strict_channel_cache[package_name]
    KeyError: 'ipython'
    
    During handling of the above exception, another exception occurred:
    
    Traceback (most recent call last):
      File "/anaconda3/lib/python3.7/site-packages/conda/exceptions.py", line 1003, in __call__
        return func(*args, **kwargs)
      File "/anaconda3/lib/python3.7/site-packages/conda/cli/main.py", line 84, in _main
        exit_code = do_call(args, p)
      File "/anaconda3/lib/python3.7/site-packages/conda/cli/conda_argparse.py", line 82, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/anaconda3/lib/python3.7/site-packages/conda/cli/main_install.py", line 20, in execute
        install(args, parser, 'install')
      File "/anaconda3/lib/python3.7/site-packages/conda/cli/install.py", line 257, in install
        force_reinstall=context.force_reinstall or context.force,
      File "/anaconda3/lib/python3.7/site-packages/conda/core/solve.py", line 107, in solve_for_transaction
        force_remove, force_reinstall)
      File "/anaconda3/lib/python3.7/site-packages/conda/core/solve.py", line 145, in solve_for_diff
        force_remove)
      File "/anaconda3/lib/python3.7/site-packages/conda/core/solve.py", line 242, in solve_final_state
        ssc = self._run_sat(ssc)
      File "/anaconda3/lib/python3.7/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/anaconda3/lib/python3.7/site-packages/conda/core/solve.py", line 475, in _run_sat
        conflicting_specs = ssc.r.get_conflicting_specs(tuple(final_environment_specs))
      File "/anaconda3/lib/python3.7/site-packages/conda/resolve.py", line 852, in get_conflicting_specs
        reduced_index = self.get_reduced_index(specs)
      File "/anaconda3/lib/python3.7/site-packages/conda/common/io.py", line 88, in decorated
        return f(*args, **kwds)
      File "/anaconda3/lib/python3.7/site-packages/conda/resolve.py", line 445, in get_reduced_index
        reduced = filter_group([s])
      File "/anaconda3/lib/python3.7/site-packages/conda/resolve.py", line 371, in filter_group
        sole_source_channel_name = self._get_strict_channel(name)
      File "/anaconda3/lib/python3.7/site-packages/conda/resolve.py", line 322, in _get_strict_channel
        all_channel_names = set(prec.channel.name for prec in self.groups[package_name])
    KeyError: 'ipython'

`$ /anaconda3/bin/conda install --name channel-priority-strict-and-pinned-file requests`

  environment variables:
                 CIO_TEST=<not set>
                CONDA_EXE=/anaconda3/bin/conda
               CONDA_ROOT=/anaconda3
              CONDA_SHLVL=0
                     PATH=/anaconda3/condabin:/Users/yhuai/dev/apache-maven-3.5.3//bin:/Library/
                          Java/JavaVirtualMachines/jdk1.8.0_131.jdk/Contents/Home/bin:/Users/yhu
                          ai/dev/scala-2.11.12//bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>

     active environment : None
            shell level : 0
       user config file : /Users/yhuai/.condarc
 populated config files : /Users/yhuai/.condarc
          conda version : 4.6.11
    conda-build version : 3.17.8
         python version : 3.7.1.final.0
       base environment : /anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/osx-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /anaconda3/pkgs
                          /Users/yhuai/.conda/pkgs
       envs directories : /anaconda3/pkgs
                          /Users/yhuai/.conda/pkgs
                          /tmp
                          /anaconda3/envs
                          /Users/yhuai/.conda/envs
               platform : osx-64
             user-agent : conda/4.6.11 requests/2.21.0 CPython/3.7.1 Darwin/18.5.0 OSX/10.14.4
                UID:GID : 502:20
             netrc file : /Users/yhuai/.netrc
           offline mode : False


An unexpected error has occurred. Conda has prepared the above report.
```
4. You can install ipython, i.e. `conda install --name channel-priority-strict-and-pinned-file ipython` works without any issue.

5. If the pinned file has an installed package, e.g. `echo "pip ==19.0.3" > /anaconda3/pkgs/channel-priority-strict-and-pinned-file/conda-meta/pinned`, `conda install --name channel-priority-strict-and-pinned-file requests` works as expected.

## Expected Behavior

When channel_priority is strict, conda install should work even pinned file has an uninstalled package. 

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : None
            shell level : 0
       user config file : /Users/yhuai/.condarc
 populated config files : /Users/yhuai/.condarc
          conda version : 4.6.11
    conda-build version : 3.17.8
         python version : 3.7.1.final.0
       base environment : /anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/osx-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/osx-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/osx-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /anaconda3/pkgs
                          /Users/yhuai/.conda/pkgs
       envs directories : /anaconda3/pkgs
                          /Users/yhuai/.conda/pkgs
                          /tmp
                          /anaconda3/envs
                          /Users/yhuai/.conda/envs
               platform : osx-64
             user-agent : conda/4.6.11 requests/2.21.0 CPython/3.7.1 Darwin/18.5.0 OSX/10.14.4
                UID:GID : 502:20
             netrc file : /Users/yhuai/.netrc
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
==> /Users/yhuai/.condarc <==
envs_dirs:
  - /anaconda3/pkgs
  - /Users/yhuai/.conda/pkgs
  - /tmp
ssl_verify: True
channel_priority: strict
channels:
  - defaults
default_channels:
  - https://repo.anaconda.com/pkgs/main/
  - https://repo.anaconda.com/pkgs/free/
  - https://repo.anaconda.com/pkgs/r/
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at /anaconda3/pkgs/channel-priority-strict-and-pinned-file:
#
# Name                    Version                   Build  Channel
ca-certificates           2019.1.23                     0    defaults
certifi                   2019.3.9                 py37_0    defaults
libcxx                    4.0.1                hcfea43d_1    defaults
libcxxabi                 4.0.1                hcfea43d_1    defaults
libedit                   3.1.20181209         hb402a30_0    defaults
libffi                    3.2.1                h475c297_4    defaults
ncurses                   6.1                  h0a44026_1    defaults
openssl                   1.1.1b               h1de35cc_1    defaults
pip                       19.0.3                   py37_0    defaults
python                    3.7.3                h359304d_0    defaults
readline                  7.0                  h1de35cc_5    defaults
setuptools                40.8.0                   py37_0    defaults
sqlite                    3.27.2               ha441bb4_0    defaults
tk                        8.6.8                ha441bb4_0    defaults
wheel                     0.33.1                   py37_0    defaults
xz                        5.2.4                h1de35cc_4    defaults
zlib                      1.2.11               h1de35cc_3    defaults
```
</p></details>
